drivers/include/periph/comms_bus.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2020 Pieter du Preez <pdupreez@gmail.com>
+ *               2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_comms_bus (I2C/SPI)
+ * @ingroup     drivers_periph
+ * @brief       Low-level I2C/SPI bus driver interface
+ *
+ * This is a kind of router for drivers that are able to support
+ * either I2C or API bus interfaces.
+ *
+ * It is required to call the comms_bus_setup function as start-up.
+ * This function sets up function pointers for directing bus
+ * communication to the desired bus.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C/SPI bus driver interface definition
+ *
+ * @author      Pieter du Preez <pdupreez@gmail.com>
+ */
+
+#ifndef PERIPH_COMMS_BUS_H
+#define PERIPH_COMMS_BUS_H
+
+/**
+ * The following check breaks the CI build.
+ * Strictly, the check is not needed here, so it's commented out for now.
+
+#if !defined(MODULE_PERIPH_I2C) && !defined(MODULE_PERIPH_SPI)
+#error At least one of the following modules are required: periph_i2c, periph_spi
+#endif
+
+*/
+
+#ifdef MODULE_PERIPH_SPI
+#include "spi.h"
+#endif
+
+#ifdef MODULE_PERIPH_I2C
+#include "i2c.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Supported bus types.
+ */
+enum comms_bus_type {
+    UNDEF = 0,
+    I2C,
+    SPI,
+};
```

Comment:
The naming of `comms_bus_type_t`  members are way too common and clashes with defines in other projects.
It breaks the CI build.

Reply:
You could namespace them with `COMMS_BUS_`. Example: `COMMS_BUS_I2C`. Why use a use `s` at the end of comms by the way ? I think `comm` is already ok.

Reply:
or maybe even `common` ?

Reply:
@aabadie: Thanks for your input. I had PERIPH_COMMS_BUS in mind. Will this be OK?

I chose `comms_bus` as an abbreviation for `communications bus`. I'm really open for anything. Let me rather change it now than later, if needed.

Would `common_bus` be better?

Reply:
@aabadie: I followed your advice and renamed `comms_bus` to `common_bus`.
I also prefixed the `common_bus_type_t` members with `COMMON_BUS_`.
The documentation is updated to the point where doxygen is happy and the CI is passing.

Is there anything else that you would like me to do for this PR?